src/components/AuditLogsTable/AuditLogsTableFilters/AuditLogsTableFilters.tsx

```diff
@@ -0,0 +1,71 @@
+import { Button, Chip, Icon, Tooltip } from "@canonical/react-components";
+import { type ReactNode } from "react";
+
+import ActionBar from "components/ActionBar/ActionBar";
+import { useQueryParams } from "hooks/useQueryParams";
+import type { SetParams } from "hooks/useQueryParams";
+
+export type AuditLogFilters = {
+  after: string | null;
+  before: string | null;
+  user: string | null;
+  model: string | null;
+  facade: string | null;
+  method: string | null;
+  version: string | null;
+};
+
+export const DEFAULT_AUDIT_LOG_FILTERS = {
+  after: null,
+  before: null,
+  user: null,
+  model: null,
+  facade: null,
+  method: null,
+  version: null,
+};
+
+export const generateFilters = (
+  filters: AuditLogFilters,
+  setFilters: SetParams<AuditLogFilters>
+) =>
+  // Reduce to remove filters that are not set.
+  Object.entries(filters).reduce<ReactNode[]>((chips, [filter, value]) => {
+    if (value) {
+      chips.push(
+        <Chip
+          className="u-no-margin--bottom"
+          key={filter}
+          lead={filter}
+          value={value}
+          onDismiss={() => setFilters({ [filter]: null })}
+        />
+      );
+    }
+    return chips;
+  }, []);
+
+const AuditLogsTableFilters = () => {
+  const [filters, setFilters] = useQueryParams<AuditLogFilters>({
+    // Spread so that the default doesn't get updated when the filters change.
+    ...DEFAULT_AUDIT_LOG_FILTERS,
+  });
+  const hasFilters = Object.values(filters).some((filter) => !!filter);
```

Comment:
I don't think this is an issue, but rather it is a small inconsistency. In case we remove all the filters manually, the `queryParams` would still leak into the url without any values. At the same time, if we remove the filters by pressing the `Clear` button, the `queryParams` would not leak into the url. I think that adding the following `useEffect` here would fix the aforementioned inconsistency:

```
 useEffect(() => {
    if (!hasFilters) {
      setFilters(DEFAULT_AUDIT_LOG_FILTERS);
    }
  }, [hasFilters, setFilters]);
```

Reply:
Thanks, it turns out that these blank query params were appearing because when I apply the filters I use the form values which are empty strings and `useQueryParams` will set them in the URL unless they are `null` or `undefined`.

I've updated it to pass `null` for blank filters and added a couple of tests.